### PR TITLE
David/pdfreports

### PR DESCRIPTION
**Pull request by DavidSarlle**
This PR implements:

- Automatic resizing of plotting area when multiple figures are returned in a plotting function
- Added fitting report in a styled html table instead of using hover property of plotly link
- Restyled fitting area and refresh rate area in plots 
- Added funcionality for exporting into pdf file or print any report
- Minor fixes intemplate html styling  

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

### CODE REVIEW

**Comment by DavidSarlle** [comment]
> @DavidSarlle I was looking into this, may I ask what is the motivation to generate PDF reports given that qibocal provides tools to store reports and share URL links?

@scarrazza Some of the advantages are:
1. It gives the users the possibilty of exporting/saving/print reports directly from the qibocal GUI
2. Adds extra funcionlity for sharing to qibocal
3. The funcionality gives a user-frinedly option for sharing reports, without using qq-export
4. With this funcionality, the user can generate reports even if the calibration routine has no finished (useful if the routine crasesh before ending)
5. Report integration into larger reports can be done more easily with pdf documents than with URLs

These are the reasons that have motivated the implementation of this functionality. Talking with fabrication team and experimentalist at TII, they were happy with this possibility.

File: src/qibocal/web/app.py [hunk 1]
```diff
@@ -10,6 +11,9 @@
 from qibocal.data import DataUnits
 from qibocal.web.server import server
 
+# import pandas as pd
+# from collections import OrderedDict
+
```
Comment (andrea-pasquale) [inline src/qibocal/web/app.py:16]: We can remove this if they are no longer needed.

File: src/qibocal/web/app.py [hunk 2]
```diff
@@ -117,16 +113,60 @@ def get_graph(interval, url, value):
...
+                                html.Th(className="th_styles", children="Value"),
+                            ]
+                        )
+                    ]
+                )
+            ]
+            table_rows = []
+
+            for i in range(0, len(fitting_params), 3):
+                # print(fitting_params[i])
```
Comment (andrea-pasquale) [inline src/qibocal/web/app.py:140]: This line can also be removed.

**Review by andrea-pasquale — COMMENTED** [review]
Thanks @DavidSarlle.
I like the table that you implemented to display the results from the fitting.
I think that the pdf report is not working as it should:
image
Can you reproduce this error?

**Comment by andrea-pasquale** [comment]
I see also that this PR is linked to #229, however I don't think that this issue is addressed in this PR, correct?

File: src/qibocal/web/app.py (hunk 1 above)
Comment (DavidSarlle) [reply src/qibocal/web/app.py]: Done

File: src/qibocal/web/app.py (hunk 2 above)
Comment (DavidSarlle) [reply src/qibocal/web/app.py]: Done

**Comment by DavidSarlle** [comment]
> I see also that this PR is linked to #229, however I don't think that this issue is addressed in this PR, correct?

It was somehow related because the possibility to export or print at any point of the execution of a calibration routine could be exported at that point. So, the user has the possibility to print/save an intermediate report. But, for sure is not automatic. the implementation in #245 is the best option.

**Comment by DavidSarlle** [comment]
@andrea-pasquale thanks for the comment. I've been able to reproduce the problem that you pointed out. I have modified the css @media print, so all the elements in the report fit in a A4 portarit orientation (and, of course in a A4 landscape). Fitting the iframes that allocates the content from the dask app (the figures and the fitting data) for all resolutions is not easy. I'have done my best, and now the reports are good enought for almost all resolutions. 
I have also elimnated some info from the pdf report to be able to show more than one figures per page. basically I have removed the summary. See below examples for typical resolutions:

width: 1920px
image

width: 1366px
image

**Comment by DavidSarlle** [comment]
@andrea-pasquale @scarrazza the PR can also be included in the first release if you are happy with the implementation. The most importat thing included in this PR is the auto resizing of the plotting area and the way of showing the fitting data. I think that is interersting if we include the PR in the first release.

**Comment by scarrazza** [comment]
Looks good to me, @andrea-pasquale let me know.

**Comment by andrea-pasquale** [comment]
Thanks @DavidSarlle. LGTM now.

**Review by andrea-pasquale — APPROVED** [review]

**Comment by andrea-pasquale** [comment]
@scarrazza when can merge it now. Tests are failing only for coverage of the plottng methods which currently we don't test.

**Pull request merged**